recipes/libtorch/all/conanfile.py

```diff
@@ -0,0 +1,667 @@
+        tc.variables["USE_GFLAGS"] = self.options.with_gflags
+        tc.variables["USE_GLOG"] = self.options.with_glog
+        tc.variables["USE_LITE_PROTO"] = self.dependencies["protobuf"].options.lite
+        tc.variables["USE_MAGMA"] = self.options.get_safe("with_magma", False)
+        tc.variables["USE_PYTORCH_METAL"] = self.options.get_safe("with_metal", False)
+        tc.variables["USE_PYTORCH_METAL_EXPORT"] = self.options.get_safe("with_metal", False)
+        tc.variables["USE_NATIVE_ARCH"] = False
+        tc.variables["USE_MPS"] = self.options.get_safe("with_mps", False)
+        tc.variables["USE_NCCL"] = self.options.get_safe("with_nccl", False)
+        tc.variables["USE_RCCL"] = self.options.get_safe("with_nccl", False) and self.options.get_safe("with_rocm", False)
+        tc.variables["USE_STATIC_NCCL"] = False
+        tc.variables["USE_SYSTEM_NCCL"] = True
+        tc.variables["USE_NNAPI"] = self.options.get_safe("with_nnapi", False)
+        tc.variables["USE_NNPACK"] = self.options.get_safe("with_nnpack", False)
+        tc.variables["USE_NUMA"] = self.options.get_safe("with_numa", False)
+        tc.variables["USE_NVRTC"] = self.options.get_safe("with_nvrtc", False)
+        tc.variables["USE_NUMPY"] = False
+        tc.variables["USE_OBSERVERS"] = self.options.observers
+        tc.variables["USE_OPENCL"] = self.options.with_opencl
+        tc.variables["USE_OPENMP"] = self.options.with_openmp
+        tc.variables["USE_PROF"] = False  # requires htrace
+        tc.variables["USE_PYTORCH_QNNPACK"] = self.options.get_safe("with_qnnpack", False)
+        tc.variables["USE_SNPE"] = self.options.get_safe("with_snpe", False)
+        tc.variables["USE_SYSTEM_EIGEN_INSTALL"] = True
+        tc.variables["USE_VALGRIND"] = False
+        tc.variables["USE_VULKAN"] = self.options.with_vulkan
+        tc.variables["USE_VULKAN_FP16_INFERENCE"] = self.options.get_safe("vulkan_fp16_inference", False)
+        tc.variables["USE_VULKAN_RELAXED_PRECISION"] = self.options.get_safe("vulkan_relaxed_precision", False)
+        tc.variables["USE_XNNPACK"] = self.options.with_xnnpack
+        tc.variables["USE_ITT"] = self.options.with_itt
+        tc.variables["USE_MKLDNN"] = self.options.get_safe("with_mkldnn", False)
+        tc.variables["USE_MKLDNN_CBLAS"] = False  # This option is useless for libtorch
+        tc.variables["USE_DISTRIBUTED"] = False  # TODO: self.options.distributed
+        tc.variables["ONNX_ML"] = True
+        tc.variables["HAVE_SOVERSION"] = True
+        tc.variables["USE_CCACHE"] = False
+        tc.variables["DEBUG_CUDA"] = self.options.get_safe("debug_cuda", False)
+        tc.variables["USE_COREML_DELEGATE"] = self.options.coreml_delegate
+        tc.variables["BUILD_LAZY_TS_BACKEND"] = self.options.build_lazy_ts_backend
+        tc.variables["USE_MIMALLOC"] = self.options.with_mimalloc
+
+        tc.variables["BLAS"] = self._blas_cmake_option_value
+
+        tc.variables["MSVC_Z7_OVERRIDE"] = False
+
```

Comment:
Incidentally, this also needs 
```
tc.variables["CMAKE_CXX_EXTENSIONS"] = True
```

Tested this while running a build that uses a compiler.cppstd. If it is using a non-gnu standard (which for other packages it must be) ATen breaks with the same error as: https://github.com/pytorch/QNNPACK/issues/67

This converts cppstd=c++17 for example to cppstd=gnu++17